File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [0]*n
    def dfs(node):
        visited[node]=1
        for i in range(n):
            if computers[node][i]==1 and i!=node and visited[i]==0:
                dfs(i)
        return
    for i in range(n):
        if visited[i]==0:
            dfs(i)
            answer+=1
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] visited = new boolean[n];
        for (int i=0; i<n; i++){
            if (!visited[i]){
                answer++;
                dfs(i, n, computers, visited);
            }
            
        }
        return answer;
    }
    
    private void dfs(int curr, int n, int[][] com, boolean[] visited){
        visited[curr] = true;
        for (int next=0; next < n; next++){
            if (com[curr][next]==1 && !visited[next]){
                dfs(next, n, com, visited);
            }
        }
    }
}